src/metatrain/utils/augmentation.py

```diff
@@ -158,12 +248,146 @@ def _apply_wigner_D_matrices(
     )
 
 
+@torch.jit.script
+def _get_system_split_indices(
+    systems: List[System],
+    sample_kind: str,
+    atom_types: List[int],
+    s2_pi: int,
+) -> List[int]:
+    """
+    Finds the indices that splits a TensorBlock along the samples axis by system index.
```

Comment:
Is this an implementation of https://github.com/metatensor/metatensor/issues/627?

Reply:
Yes :)